src/components/Hoverable/index.js

```diff
@@ -1,189 +1,198 @@
-            child = this.props.children[0];
+    useEffect(() => {
+        if (disabled) {
+            return;
         }
-
-        if (_.isFunction(child)) {
-            child = child(this.state.isHovered);
+        if (onHoverIn && isHovered) {
+            return onHoverIn();
         }
-
-        if (!DeviceCapabilities.hasHoverSupport()) {
-            return child;
+        if (onHoverOut && !isHovered) {
+            return onHoverOut();
         }
-
-        return React.cloneElement(React.Children.only(child), {
-            ref: (el) => {
-                this.wrapperView = el;
-
-                // Call the original ref, if any
-                const {ref} = child;
-                if (_.isFunction(ref)) {
-                    ref(el);
-                    return;
-                }
-
-                if (_.isObject(ref)) {
-                    ref.current = el;
-                }
-            },
-            onMouseEnter: (el) => {
-                this.setIsHovered(true);
-
-                if (_.isFunction(child.props.onMouseEnter)) {
-                    child.props.onMouseEnter(el);
-                }
-            },
-            onMouseLeave: (el) => {
-                this.setIsHovered(false);
-
-                if (_.isFunction(child.props.onMouseLeave)) {
-                    child.props.onMouseLeave(el);
-                }
-            },
-            onBlur: (el) => {
-                // Check if the blur event occurred due to clicking outside the element
-                // and the wrapperView contains the element that caused the blur and reset isHovered
-                if (!this.wrapperView.contains(el.target) && !this.wrapperView.contains(el.relatedTarget)) {
-                    this.setIsHovered(false);
-                }
-
-                if (_.isFunction(child.props.onBlur)) {
-                    child.props.onBlur(el);
-                }
-            },
-        });
+    }, [disabled, isHovered, onHoverIn, onHoverOut]);
+
+    // Expose inner ref to parent through outerRef. This enable us to use ref both in parent and child.
+    useImperativeHandle(outerRef, () => ref.current, []);
+
+    const child = useMemo(() => React.Children.only(mapChildren(children, isHovered)), [children, isHovered]);
+
+    const onMouseEnter = useCallback(
+        (el) => {
+            updateIsHoveredOnScrolling(true);
+
+            if (_.isFunction(child.props.onMouseEnter)) {
+                child.props.onMouseEnter(el);
+            }
+        },
+        [child.props, updateIsHoveredOnScrolling],
+    );
+
+    const onMouseLeave = useCallback(
+        (el) => {
+            updateIsHoveredOnScrolling(false);
+
+            if (_.isFunction(child.props.onMouseLeave)) {
+                child.props.onMouseLeave(el);
+            }
+        },
+        [child.props, updateIsHoveredOnScrolling],
+    );
+
+    const onBlur = useCallback(
+        (el) => {
```

Comment:
Let's rename to use normal names.
```suggestion
        (element) => {
```

Reply:
What's the rule behind this suggestion? The `el` name was kept from previous implementation. AFAIK, most of the callbacks uses `el` as a name instead of `element`

Reply:
I am not sure about the rule within some Expensify's docs. I see it is like from the general perspective, general recommendations to avoid short names and use normal (_full_) names for variables. 

I understand that `el` name was kept from previous implementation. I would say that's up to you.

Reply:
I based the decision on what is most widely used in the codebase in such scenarios (let's serach `(el)` vs `(element)` in the codebase to see the results). Thanks for the suggestion and detailed review! 👍 

P.S. 

As a rule of thumb, I'd say that for lambda/inline functions with clear responsibility, it's okay to make names shorter. For example:
```jsx
vegetables.forEach(v => v.eat())
```

Reply:
Yeah, also makes sense.